### DISCUSSION TITLE
How to configure Hardware-Shutter-Button

### DISCUSSION THREAD

**Discussion by MMeinhardt1** [arendst/Tasmota#16560] [Q&A]
Hello community,

I sometimes have my problems with the descriptions of the functions in tasmota.
Therefore I have to ask here now again exactly and would like a "step by step" description.
I have a Loratap SC500W-V2 shutter switch with tasmota V12.0.1, the control via ioBroker over mqtt works without problems. Now I would like to connect shutter buttons to the designated inputs on the SC500W.
Can someone please tell me what exactly I need to activate and set in tasmota? I am unfortunately somehow too stupid to understand the descriptions correctly.

Answers may also be written in German.

Thank you and have a nice Sunday!

EDIT: Is it possible to configure it like this?
press / short press (<1 sec.) - open / close completely
press / long press (>1 sec.) - moves as long as is pressed

**Comment by stefanbode** [comment]
Hello @MMeinhardt1 : Lets assume you already overcome all configurations with setoption81 the shutteropenduration and so on and the shutter is working, but NOW you need the buttons.
In the configuration you define a BUTTON 1 and a BUTTON 2. Please do NOT use SWITCHES. This will not work.
Now you define the shutteroperation to your button and here you will get the three lines you have to enter on command line.
```
SetOption32 5
ShutterButton1 1  up 1
ShutterButton1 2  down 1
```

Have fun

**Reply by MMeinhardt1** [reply]
Hi @stefanbode,
currently I am in Vancouver for work. After I will back at home I try your solution and give a feedback. 

Many thanks

**Reply by miazza99** [reply]
> Hello @MMeinhardt1 : Lets assume you already overcome all configurations with setoption81 the shutteropenduration and so on and the shutter is working, but NOW you need the buttons. In the configuration you define a BUTTON 1 and a BUTTON 2. Please do NOT use SWITCHES. This will not work. Now you define the shutteroperation to your button and here you will get the three lines you have to enter on command line.
> 
> ```
> SetOption32 5
> ShutterButton1 1  up 1
> ShutterButton1 2  down 1
> ```
> 
> Have fun

I tried the same on my SonOFF Dual R3 with Tasmota 15.3.0 but my buttons remains without any behaviour change.
Below is my settings.
Now when I push the buttom down it start but as soon as I depress it it stops (the same with up).

`
18:45:23.209 CMD: SetOption32 5
18:45:23.215 MQT: stat/Tapparella_Sala_Sx/RESULT = {"SetOption32":5}
18:46:22.845 CMD: ShutterButton1 1
18:46:22.848 SHT: About to save settings to file /.drvset027
18:46:22.910 MQT: stat/Tapparella_Sala_Sx/RESULT = {"ShutterButton1":"1 0/0 0/0 0/0 0/0 - - - -|2 0 1 25 0 - - - -"}
18:47:13.795 CMD: ShutterButton1 2  down 1
18:47:13.801 MQT: stat/Tapparella_Sala_Sx/RESULT = {"ShutterButton1":"1 0/0 0/0 0/0 0/0 - - - -|2 0 1 25 0 - - - -"}
`

**Comment by caprez95** [comment]
Hello together

I have some problems with the Shutter Buttons.
I defined two Buttons for 1 Shutter as follow:

ShutterButton1 1 - - - 100..0 - - - - -
ShutterButton1 2 - - - 0..100 - - - - -

That means:
MultiPress (Single, Double and Triple-Press) deactivated and only ButtonHold active

<img width="394" alt="ShutterButton_Advanced" src="https://github.com/arendst/Tasmota/assets/45364281/1ccbfddf-7725-4191-8d98-95b4ef84df3b">

Thats works fine. MultiPress (up to triple-press) is ignored and only on Button Hold the Shutter moves, and on Button release the Shutter Stopps.

BUT, unfortunately the ShutterButton reacts to a Quadruple-Press.
After a Quadruple-Press, the relay remains on permanently until the upper or lower end position of the shutter is reached (see video).

https://github.com/arendst/Tasmota/assets/45364281/ba5f95cd-fca5-4578-950b-6857e72b3734

But the quadruple-Press cannot be defined for a shutter button.

Does anyone know where this comes from?
Does this come from the button multi-press functions? 
Are they active on the shutter buttons? 
And can the multi-press functions be deactivated?

Many thanks for every answer.
Kind regards Carlo

**Reply by stefanbode** [reply]
Just for the hold action define - and you are fine. 100..0 is not a valid option. You can put in a value e.g 66 and then hold will go to 66%.

**Reply by caprez95** [reply]
Yes I know. For the first Shutter Button I will fully close (100) and for the other fully opened (0).
But thats not the problem. The Hold Action works fine.

My problem is the behavior of the Quad-Press.
This I can't define in the ShutterButton Options.

**Reply by stefanbode** [reply]
Yes. You also cannot define penta press or something else. Three presses is the max.

**Reply by stefanbode** [reply]
I will check what function causes the relay to execute on 4-times press. This is not a shutter function. Seems to come from the outside.

**Reply by stefanbode** [reply]
ok, I got it and commit a change. on QUAD press the HOLD event was executed. This is for sure a regression. fixed. Fixed by change: #19589

**Reply by caprez95** [reply]
Thank you very much for your research.
The problem doesn't just seem to be with the quadruple click. 
5 or more clicks are also accepted with the same function.
<img width="849" alt="Unbenannt" src="https://github.com/arendst/Tasmota/assets/45364281/5b899cd3-bde2-4887-8753-f27349ef5a94">

**Reply by stefanbode** [reply]
This works different here. 4 and more clicks gets rerouted back to the multi press events. There is no action on the shutter. I can press 4 or five times and there is no shutter movement. Can you share a log with loglevel 4. Of the 5 times press event? Before my fix the 4 press where accidentally recognize as hold event.

**Reply by stefanbode** [reply]
The image you show is correct. The shutter do nothing. Just continue. This is intended. Button1 5 is correct. This is the recognized event.

**Reply by caprez95** [reply]
For me, 5 or 6 clicks have exactly the same effect as 4 clicks.
Shutter close to lower end position and only stops with another click.

**Reply by stefanbode** [reply]
Can you share a log from stop state and then 5 presses. Thx

**Reply by caprez95** [reply]
06:58:50.077 HTP: Consoles
06:58:51.438 HTP: Console
06:58:58.697 WIF: Sending Gratuitous ARP
06:59:08.472 WIF: Checking connection...
06:59:13.137 SRC: Shutter
06:59:13.139 SHT: FUNC_SET_POWER Relaymask 0 SwitchedRelay:1 by Shutter, payload 100, powermask 1
06:59:13.141 SHT: FUNC_SET_POWER end. powermask 1
06:59:13.146 RSL: RESULT = {"POWER1":"ON"}
06:59:13.148 RSL: POWER1 = ON
06:59:13.150 SHT: Shtr1 Real 106000, Start 106000, Stop 200000, Dir 1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
06:59:13.154 RSL: RESULT = {"Shutter1":{"Position":53,"Direction":1,"Target":100,"Tilt":0},"Shutter2":{"Position":50,"Direction":0,"Target":50,"Tilt":0}}
06:59:13.160 RSL: RESULT = {"Shutter1":{"Button1":5}}
06:59:13.166 SHT: Shtr1 Time 1(1), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 106000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.223 SHT: Shtr1 Time 2(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 107000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.272 SHT: Shtr1 Time 3(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 108000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.322 SHT: Shtr1 Time 4(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 109000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.373 SHT: Shtr1 Time 5(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 110000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.424 SHT: Shtr1 Time 6(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 111000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.474 SHT: Shtr1 Time 7(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 112000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.525 SHT: Shtr1 Time 8(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 113000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.574 SHT: Shtr1 Time 9(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 114000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.625 SHT: Shtr1 Time 10(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 115000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.676 SHT: Shtr1 Time 11(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 116000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.679 SHT: Shtr1 Real 116000, Start 106000, Stop 200000, Dir 1, Delay 0, Rtc 0.55 [s], Freq 0, PWM 0, Tilt 0
06:59:13.683 RSL: RESULT = {"Shutter1":{"Position":58,"Direction":1,"Target":100,"Tilt":0},"Shutter2":{"Position":50,"Direction":0,"Target":50,"Tilt":0}}
06:59:13.726 SHT: Shtr1 Time 12(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 117000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.776 SHT: Shtr1 Time 13(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 118000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.826 SHT: Shtr1 Time 14(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 119000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.876 SHT: Shtr1 Time 15(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 120000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.925 SHT: Shtr1 Time 16(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 121000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:13.976 SHT: Shtr1 Time 17(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 122000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:14.026 SHT: Shtr1 Time 18(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 123000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.376 SHT: Shtr1 Time 45(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 150000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.426 SHT: Shtr1 Time 46(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 151000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.477 SHT: Shtr1 Time 47(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 152000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.526 SHT: Shtr1 Time 48(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 153000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.576 SHT: Shtr1 Time 49(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 154000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.626 SHT: Shtr1 Time 50(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 155000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.676 SHT: Shtr1 Time 51(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 156000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.680 SHT: Shtr1 Real 156000, Start 106000, Stop 200000, Dir 1, Delay 0, Rtc 2.55 [s], Freq 0, PWM 0, Tilt 0
06:59:15.684 RSL: RESULT = {"Shutter1":{"Position":78,"Direction":1,"Target":100,"Tilt":0},"Shutter2":{"Position":50,"Direction":0,"Target":50,"Tilt":0}}
06:59:15.727 SHT: Shtr1 Time 52(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 157000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.777 SHT: Shtr1 Time 53(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 158000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.827 SHT: Shtr1 Time 54(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 159000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.877 SHT: Shtr1 Time 55(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 160000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.928 SHT: Shtr1 Time 56(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 161000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:15.977 SHT: Shtr1 Time 57(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 162000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:16.028 SHT: Shtr1 Time 58(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 163000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:16.077 SHT: Shtr1 Time 59(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 164000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:16.128 SHT: Shtr1 Time 60(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 165000, nStop 0, Trgt 200000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:16.177 SHT: Shtr1, Button 1 = 1 (single=1, double=2, tripple=3, hold=4)
06:59:16.179 SHT: Stop 1: dir: 1
06:59:16.181 SHT: Shtr1 Time 61(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 166000, nStop 0, Trgt 166000, mVelo 0, Dir 1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
06:59:16.183 SHT: Stop 1 Mode 0 time 61
06:59:16.185 SRC: Shutter
06:59:16.187 SHT: FUNC_SET_POWER Relaymask 1 SwitchedRelay:1 by Shutter, payload 1, powermask 0
06:59:16.189 SHT: FUNC_SET_POWER end. powermask 0
06:59:16.194 RSL: RESULT = {"POWER1":"OFF"}
06:59:16.196 RSL: POWER1 = OFF
06:59:16.199 SHT: Shtr1 Real 166000, Start 166000, Stop 166000, Dir 0, Delay 0, Rtc 3.05 [s], Freq 0, PWM 0, Tilt 0
06:59:16.201 RSL: SHUTTER1 = 83
06:59:16.204 RSL: RESULT = {"Shutter1":{"Position":83,"Direction":0,"Target":83,"Tilt":0}}
06:59:16.212 RSL: RESULT = {"Shutter1":{"Button1":1}}
06:59:16.624 CFG: Saved to flash at 3FB, Count 38, Bytes 4096

**Reply by caprez95** [reply]
Do you have any news on this?

**Reply by stefanbode** [reply]
From the log I see there is not button evolvement at all. The relay is directly triggered by the button. Maybe something is missing in the beginning, but there is no value able info available.

**Reply by stefanbode** [reply]
The stop is totally correct. At the start relay1 is triggered for an unknown reason.

**Reply by caprez95** [reply]
Ok strange. Anyway many thanks for your time.

**Reply by stefanbode** [reply]
06:59:13.139 SHT: FUNC_SET_POWER Relaymask 0 SwitchedRelay:1

This should not happen. Button1 5 times should not trigger the relay. Buttons always execute a Shutterposition and this triggers the relay.
